src/__tests__/integration/userInputValidation.test.ts

```
@@ -0,0 +1,445 @@
+        const result = sanitizeShareFilename(name);
+        
+        // Should detect that original was unsafe
+        expect(result.originalUnsafe || result.errors.length > 0).toBe(true);
+        
+        // Sanitized version should be safe for filesystem
+        expect(result.sanitized).toBeTruthy();
+        expect(result.sanitized.length).toBeLessThanOrEqual(FILESYSTEM_LIMITS.MAX_FILENAME_LENGTH);
+        
+        // Should not contain unsafe characters
+        // eslint-disable-next-line no-control-regex
+        expect(result.sanitized).not.toMatch(/[<>:"/\\|?*\x00-\x1f]/);
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)